### PR TITLE
Run ViT examples on PyTorch XLA out of the box

### PR DESCRIPTION
**Pull request by miladm**
Check if Vision Transformers (ViT) model and its example script are supported on PyTorch/XLA out of the box. If not, produce the code to bring the needed support.

### CODE REVIEW

File: examples/pytorch/image-classification/run_image_classification.py [hunk 1]
```diff
@@ -207,6 +207,7 @@ def main():
         data_files=data_args.data_files,
         cache_dir=model_args.cache_dir,
         task="image-classification",
+        ignore_verifications=True,
```
Comment (sgugger) [inline examples/pytorch/image-classification/run_image_classification.py:210]: I don't think this should be added here. This is not in any other examples, why would we need it here?

**Review by sgugger — COMMENTED** [review]
Thanks for adding this! For the quality check, you need to install the quality dependencies with `pip install[quality]` then run `make style` to properly format your code.

File: examples/pytorch/image-classification/run_image_classification.py (hunk 1 above)
Comment (miladm) [reply examples/pytorch/image-classification/run_image_classification.py:210]: That's a fair point. It turns out the verification stage causes the following code crash. 

It also turns out the the ViT test completes successfully when `ignore_verifications=True`, and so I decided to add this flag. 

What is your recommendation  @sgugger?

My repro command for reference:
`python3 examples/pytorch/xla_spawn.py --num_cores 1 examples/pytorch/image-classification/run_image_classification.py   --dataset_name cats_vs_dogs     --output_dir ./cats_vs_dogs_outputs/     --remove_unused_columns False     --do_train     --do_eval  --learning_rate 2e-4     --num_train_epochs 5     --per_device_train_batch_size 32     --per_device_eval_batch_size 32     --logging_strategy steps     --logging_steps 10     --evaluation_strategy epoch     --save_strategy epoch     --load_best_model_at_end True     --save_total_limit 3     --seed 1337  --overwrite_output_dir`

```
Traceback (most recent call last):
  File "examples/pytorch/xla_spawn.py", line 86, in <module>
    main()
  File "examples/pytorch/xla_spawn.py", line 82, in main
    xmp.spawn(mod._mp_fn, args=(), nprocs=args.num_cores)
  File "/usr/local/lib/python3.8/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 387, in spawn
    _start_fn(0, pf_cfg, fn, args)
  File "/usr/local/lib/python3.8/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 324, in _start_fn
    fn(gindex, *args)
  File "/home/miladmo/transformers/examples/pytorch/image-classification/run_image_classification.py", line 354, in _mp_fn
    main()
  File "/home/miladmo/transformers/examples/pytorch/image-classification/run_image_classification.py", line 204, in main
    ds = load_dataset(
  File "/home/miladmo/.local/lib/python3.8/site-packages/datasets/load.py", line 1702, in load_dataset
    builder_instance.download_and_prepare(
  File "/home/miladmo/.local/lib/python3.8/site-packages/datasets/builder.py", line 594, in download_and_prepare
    self._download_and_prepare(
  File "/home/miladmo/.local/lib/python3.8/site-packages/datasets/builder.py", line 695, in _download_and_prepare
    verify_splits(self.info.splits, split_dict)
  File "/home/miladmo/.local/lib/python3.8/site-packages/datasets/utils/info_utils.py", line 74, in verify_splits
    raise NonMatchingSplitsSizesError(str(bad_splits))
datasets.utils.info_utils.NonMatchingSplitsSizesError: [{'expected': SplitInfo(name='train', num_bytes=7503250, num_examples=23422, dataset_name='cats_vs_dogs'), 'recorded': SplitInfo(name='train', num_bytes=7452557, num_examples=23410, dataset_name='cats_vs_dogs')}]
```

Comment (sgugger) [reply examples/pytorch/image-classification/run_image_classification.py:210]: I can't reproduce on my side. @lhoestq do you have an idea of what could go wrong here?

Comment (lhoestq) [reply examples/pytorch/image-classification/run_image_classification.py:210]: It looks like the cache of this dataset is outdated. There was a [fix](https://github.com/huggingface/datasets/pull/3752) 3 weeks ago and 23410 is the right number of examples

Using `download_mode="force_redownload"` should fix this

Comment (miladm) [reply examples/pytorch/image-classification/run_image_classification.py:210]: @lhoestq how do I used `download_mode` in the execution command?

FYI: on my side, I ran ViT on a brand new TPU VM node last night. I don't expect the node had a pre-existing cache of the dataset.

Comment (miladm) [reply examples/pytorch/image-classification/run_image_classification.py:210]: @lhoestq what do you recommend I do as a next step? Shall I keep the `ignore_verifications=True` paramter?

Comment (sgugger) [reply examples/pytorch/image-classification/run_image_classification.py:210]: With the fix in datasets, it shouldn't be necessary aymore.

**Comment by miladm** [comment]
I dropped the ball here.
@steventk-g can please follow up on this PR and help land it in HF stack?

**Pull request closed without merging**